homeassistant/components/api/__init__.py

```diff
@@ -385,6 +385,44 @@ async def post(
             )
 
         context = self.context(request)
+
+        if "return_response" in request.query or (
+            data is not None
+            and isinstance(data, dict)
+            and data.pop("return_response", False)
+        ):
```

Comment:
Let's not conflate directives to the service call with service call data, it's enough with a query parameter:

```suggestion
        if "return_response" in request.query:
```

Reply:
Sure, it was mostly for flexibility: depending on what the user is doing, it might be easier for them to change the JSON payload rather than the query string.

Reply:
Done, please take a look.